crates/re_viewport/src/space_info.rs

```diff
@@ -116,18 +129,102 @@ pub struct SpaceInfoCollection {
     spaces: BTreeMap<EntityPath, SpaceInfo>,
 }
 
+impl SpaceInfoCollection {
+    pub fn len(&self) -> usize {
+        self.spaces.len()
+    }
+}
+
+// TODO(jleibs): These need to get pulled from a registry of some form
```

Comment:
also ideally they're not in `re_viewport` but this is goverened by the heuristic method on the SpaceView trait. re_viewport knowing about re_types is kind of a hack

Reply:
Associating some kind of "Preferred `SpaceInfoCategory`" metadata in the schema registry is an interesting thought.
Related:
- https://github.com/rerun-io/rerun/issues/3661#issuecomment-1754641097